storybook/utils/zeroReports.test.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * Zero Reports utility tests.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { replaceValuesOrRemoveRowForDateRangeInAnalyticsReport } from './zeroReports';
+import mockedReportResponse from '../../assets/js/modules/analytics-4/utils/__fixtures__/mocked-report-multiple-distinct-date-ranges.json';
+
+describe( 'zeroReports', () => {
+	describe( 'replaceValuesOrRemoveRowForDateRangeInAnalyticsReport', () => {
+		const matchesDateRange = ( data, dateRange ) =>
+			data.some( ( value ) => value.value === dateRange );
+
+		it( 'should zero values for date_range_0 when emptyRowBehavior is "zero"', () => {
+			// Create a deep copy of the mocked report
+			const report = JSON.parse( JSON.stringify( mockedReportResponse ) );
+			const result =
+				replaceValuesOrRemoveRowForDateRangeInAnalyticsReport(
+					report,
+					'date_range_0',
+					'zero'
+				);
+
+			// Expect all rows to be present (10 total)
+			expect( result.rows ).toHaveLength( 10 );
+
+			// Check that date_range_0 rows have zeroed metric values
+			const dateRange0Rows = result.rows.filter( ( row ) =>
+				matchesDateRange( row.dimensionValues, 'date_range_0' )
+			);
+
+			dateRange0Rows.forEach( ( row ) => {
+				row.metricValues.forEach( ( metricValue ) => {
+					expect( metricValue.value ).toBe( '0' );
+				} );
+			} );
+
+			// Check that date_range_1 rows are unchanged
+			const dateRange1Rows = result.rows.filter( ( row ) =>
+				matchesDateRange( row.dimensionValues, 'date_range_1' )
+			);
+
+			dateRange1Rows.forEach( ( row, index ) => {
+				const originalRow = report.rows.filter( ( r ) =>
+					matchesDateRange( r.dimensionValues, 'date_range_1' )
+				)[ index ];
+
+				expect( row.metricValues ).toEqual( originalRow.metricValues );
+			} );
+
+			// Check that totals for date_range_0 are zeroed
+			const totalForDateRange0 = result.totals.find( ( total ) =>
+				total.dimensionValues.some(
+					( value ) => value.value === 'date_range_0'
+				)
+			);
+
+			totalForDateRange0.metricValues.forEach( ( metricValue ) => {
+				expect( metricValue.value ).toBe( '0' );
+			} );
+
+			// Check that totals for date_range_1 are unchanged
+			const totalForDateRange1 = result.totals.find( ( total ) =>
```

Comment:
Okay, I'm going to leave my other suggestions in because it shows the variable names here are a bit unclear/ambiguous and confused me at first too. These are clearly keyed to "date range with the key `'date_range_0'`" and `"date range with the key `'date_range_1'`", but the variable names also read like ranges with a number of rows/results/data/etc.

"First date range" and "second date range", or "past date range" and "current date range" as descriptors would be clearer here, I think.